tests/components/recorder/test_purge.py

```diff
@@ -59,6 +61,14 @@ async def test_purge_old_states(
         finished = purge_old_data(instance, purge_before, repack=False)
         assert finished
         assert states.count() == 2
+        assert "test.recorder2" in instance._old_states
+
+        # run purge_old_data again
+        purge_before = dt_util.utcnow()
+        finished = purge_old_data(instance, purge_before, repack=False)
+        assert not finished
+        assert states.count() == 0
+        assert "test.recorder2" not in instance._old_states
```

Comment:
Should we record some more data and verify it ends up in the database here to have a full end to end test?

Reply:
Yes, added in 78c038a